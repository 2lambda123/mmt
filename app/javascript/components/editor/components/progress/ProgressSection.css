.section-circle {
    font-size: 2.60em; 
    display: block; 
    float: left; 
    margin-right: 10px
}

.section-label {
    cursor: pointer;
    font-size: 16px;
    font-family: sans-serif;
    padding-left: 4px
}

.pass-circle {
    color: rgb(24, 156, 84)
}

.invalid-circle {
    color: rgb(24, 156, 84)
}