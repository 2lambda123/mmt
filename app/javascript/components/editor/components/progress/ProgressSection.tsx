/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/anchor-is-valid */
/* eslint-disable react/jsx-no-comment-textnodes */
import { observer } from 'mobx-react'
import React, { ReactNode } from 'react'
import _ from 'lodash'
import { FieldInfo } from './FieldInfo'
import { ProgressCircleType } from './ProgressCircleType'
import ProgressField from './ProgressField'
import './ProgressSection.css'
import withRouter from '../withRouter'
import MetadataEditor from '../../MetadataEditor'

type ProgressSectionProps = {
  section: string
  status: ProgressCircleType
  fields: Array<FieldInfo>
  router: RouterType
  editor: MetadataEditor
}

class ProgressSection extends React.Component<ProgressSectionProps, never> {
  progressSectionCircle(type: ProgressCircleType) {
    switch (type) {
      case ProgressCircleType.Pass:
        return (
          <i className="eui-icon eui-check pass-circle" />
        )
      case ProgressCircleType.NotStarted:
      case ProgressCircleType.Error:
      case ProgressCircleType.Invalid:
        return (
          <i className="eui-icon eui-fa-circle-o invalid-circle" />
        )
      default: return (
        <span>Status Unknown</span>
      )
    }
  }
  navigateTo() {
    const {
      section, router, editor
    } = this.props
    const { params } = router
    const { id } = params
    const sectionName = section.replace(/\s/g, '_')
    setTimeout(() => {
      window.location.href = `/${editor.documentType}/${id}/edit/${sectionName}`
    }, 50)
    // navigate(`/${editor.documentType}/${id}/edit/${sectionName}`, { replace: false })
  }
  render() {
    const {
      section, status, fields, editor
    } = this.props
    const rows = fields.map((field: FieldInfo): ReactNode => (
      <span key={JSON.stringify(field)} data-testid={`progress-field__${_.kebabCase(field.name)}_${field.index}`}>
        <ProgressField fieldInfo={field} editor={editor} />
      </span>
    ))
    return (
      <div className="progress-view" key={JSON.stringify(fields)} data-testid={`progress-section__${_.kebabCase(section)}_${_.kebabCase(status)}`}>
        <div className="section-circle">
          {this.progressSectionCircle(status)}
        </div>
        <div>
          <div className="section-label">
            <div
              onClick={() => this.navigateTo()}
              data-testid={`progress-section__${_.kebabCase(section)}_link`}
            >
              {section}
            </div>
          </div>
          <div>
            {rows}
          </div>
        </div>
      </div>
    )
  }
}
export default withRouter(observer(ProgressSection))
