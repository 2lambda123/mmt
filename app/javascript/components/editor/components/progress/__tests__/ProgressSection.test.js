import React from 'react'
import {
  render
} from '@testing-library/react'
import {
  BrowserRouter
} from 'react-router-dom'
import ProgressSection from '../ProgressSection'

describe('Progress Section Test', () => {
  it('renders in case section passed', async () => {
    const props = {
      section: 'Tool Information',
      status: 'Pass',
      fields: [
        {
          section: 'Tool Information',
          name: 'Name',
          index: null,
          status: 'Pass',
          required: true
        },
        {
          section: 'Tool Information',
          name: 'LongName',
          index: null,
          status: 'Pass',
          required: true
        },
        {
          section: 'Tool Information',
          name: 'Version',
          index: null,
          status: 'Pass',
          required: true
        },
        {
          section: 'Tool Information',
          name: 'VersionDescription',
          index: null,
          status: 'Not Started',
          required: false
        },
        {
          section: 'Tool Information',
          name: 'Type',
          index: null,
          status: 'Pass',
          required: true
        },
        {
          section: 'Tool Information',
          name: 'LastUpdatedDate',
          index: null,
          status: 'Not Started',
          required: false
        },
        {
          section: 'Tool Information',
          name: 'Description',
          index: null,
          status: 'Pass',
          required: true
        },
        {
          section: 'Tool Information',
          name: 'DOI',
          index: null,
          status: 'Pass',
          required: false
        },
        {
          section: 'Tool Information',
          name: 'URL',
          index: null,
          status: 'Pass',
          required: true
        }
      ]
    }
    const { getByTestId, container } = render(
      <BrowserRouter>
        <ProgressSection {...props} />
      </BrowserRouter>
    )
    const component = getByTestId('progress-section__tool-information_pass')
    expect(component).toBeTruthy()
    expect(component).toHaveTextContent('Tool Information')
    expect(getByTestId('progress-field__name_null_required_pass')).toBeTruthy()
    expect(getByTestId('progress-field__long-name_null_required_pass')).toBeTruthy()
    expect(getByTestId('progress-field__version_null_required_pass')).toBeTruthy()
    expect(getByTestId('progress-field__version-description_null_not-required_not-started')).toBeTruthy()
    expect(getByTestId('progress-field__type_null_required_pass')).toBeTruthy()
    expect(getByTestId('progress-field__last-updated-date_null_not-required_not-started')).toBeTruthy()
    expect(getByTestId('progress-field__description_null_required_pass')).toBeTruthy()
    expect(getByTestId('progress-field__doi_null_not-required_pass')).toBeTruthy()
    expect(getByTestId('progress-field__url_null_required_pass')).toBeTruthy()
    expect(container).toMatchSnapshot()
  })
  it('renders in case section not passed', async () => {
    const props = {
      section: 'Tool Information',
      status: 'Error',
      fields: [
        {
          section: 'Tool Information',
          name: 'Name',
          index: null,
          status: 'Pass',
          required: true
        },
        {
          section: 'Tool Information',
          name: 'LongName',
          index: null,
          status: 'Error',
          required: true
        },
        {
          section: 'Tool Information',
          name: 'Version',
          index: null,
          status: 'Pass',
          required: true
        },
        {
          section: 'Tool Information',
          name: 'VersionDescription',
          index: null,
          status: 'Not Started',
          required: false
        },
        {
          section: 'Tool Information',
          name: 'Type',
          index: null,
          status: 'Pass',
          required: true
        },
        {
          section: 'Tool Information',
          name: 'LastUpdatedDate',
          index: null,
          status: 'Not Started',
          required: false
        },
        {
          section: 'Tool Information',
          name: 'Description',
          index: null,
          status: 'Pass',
          required: true
        },
        {
          section: 'Tool Information',
          name: 'DOI',
          index: null,
          status: 'Pass',
          required: false
        },
        {
          section: 'Tool Information',
          name: 'URL',
          index: null,
          status: 'Pass',
          required: true
        }
      ]
    }
    const { getByTestId, container } = render(
      <BrowserRouter>
        <ProgressSection {...props} />
      </BrowserRouter>
    )
    const component = getByTestId('progress-section__tool-information_error')
    expect(component).toBeTruthy()
    expect(component).toHaveTextContent('Tool Information')
    expect(getByTestId('progress-field__name_null_required_pass')).toBeTruthy()
    expect(getByTestId('progress-field__long-name_null_required_error')).toBeTruthy()
    expect(getByTestId('progress-field__version_null_required_pass')).toBeTruthy()
    expect(getByTestId('progress-field__version-description_null_not-required_not-started')).toBeTruthy()
    expect(getByTestId('progress-field__type_null_required_pass')).toBeTruthy()
    expect(getByTestId('progress-field__last-updated-date_null_not-required_not-started')).toBeTruthy()
    expect(getByTestId('progress-field__description_null_required_pass')).toBeTruthy()
    expect(getByTestId('progress-field__doi_null_not-required_pass')).toBeTruthy()
    expect(getByTestId('progress-field__url_null_required_pass')).toBeTruthy()
    expect(container).toMatchSnapshot()
  })
  it('renders in case section empty', async () => {
    const props = {
      section: 'Tool Contacts',
      status: 'Not Started',
      fields: [
        {
          section: 'Tool Contacts',
          name: 'ContactGroups',
          index: null,
          status: 'Not Started',
          required: true
        },
        {
          section: 'Tool Contacts',
          name: 'ContactPersons',
          index: null,
          status: 'Not Started',
          required: false
        }
      ]
    }
    const { getByTestId, container } = render(
      <BrowserRouter>
        <ProgressSection {...props} />
      </BrowserRouter>
    )
    const component = getByTestId('progress-section__tool-contacts_not-started')
    expect(component).toBeTruthy()
    expect(component).toHaveTextContent('Tool Contacts')
    expect(getByTestId('progress-field__contact-groups_null_required_not-started')).toBeTruthy()
    expect(getByTestId('progress-field__contact-persons_null_not-required_not-started')).toBeTruthy()
    expect(container).toMatchSnapshot()
  })
  it('renders in case section status unknown', async () => {
    const props = {
      section: 'Tool Contacts',
      status: 'Made Up',
      fields: [
        {
          section: 'Tool Contacts',
          name: 'ContactGroups',
          index: null,
          status: 'Not Started',
          required: true
        },
        {
          section: 'Tool Contacts',
          name: 'ContactPersons',
          index: null,
          status: 'Not Started',
          required: false
        }
      ]
    }
    const { getByTestId, container } = render(
      <BrowserRouter>
        <ProgressSection {...props} />
      </BrowserRouter>
    )
    const component = getByTestId('progress-section__tool-contacts_made-up')
    expect(component).toBeTruthy()
    expect(component).toHaveTextContent('Tool Contacts')
    expect(component).toHaveTextContent('Status Unknown')
    expect(getByTestId('progress-field__contact-groups_null_required_not-started')).toBeTruthy()
    expect(getByTestId('progress-field__contact-persons_null_not-required_not-started')).toBeTruthy()
    expect(container).toMatchSnapshot()
  })
})
