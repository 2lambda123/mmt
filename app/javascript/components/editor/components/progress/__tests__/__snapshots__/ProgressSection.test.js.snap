// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Section Test renders in case section empty 1`] = `
<div>
  <div
    data-testid="progress-section__tool-contacts_not-started"
    style="display: flex; align-items: center;"
  >
    <div
      class="section-circle"
    >
      <i
        class="eui-icon eui-fa-circle-o invalid-circle"
      />
    </div>
    <div>
      <div
        class="section-label"
      >
        <div
          data-testid="progress-section__tool-contacts_link"
        >
          Tool Contacts
        </div>
      </div>
      <div>
        <span
          data-testid="progress-field__contact-groups_null"
        >
          <span
            class="field-circle"
            data-testid="progress-field__contact-groups_null_required_not-started"
          >
            <span
              class="progressField"
              data-testid="progress-field__contact-groups_link"
            >
              <i
                class="eui-icon eui-required-o not-started-required-circle"
                title="ContactGroups is required"
              />
            </span>
          </span>
        </span>
        <span
          data-testid="progress-field__contact-persons_null"
        >
          <span
            class="field-circle"
            data-testid="progress-field__contact-persons_null_not-required_not-started"
          >
            <span
              class="progressField"
              data-testid="progress-field__contact-persons_link"
            >
              <i
                class="eui-icon eui-fa-circle-o not-started-not-required-circle"
                title="ContactPersons"
              />
            </span>
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Progress Section Test renders in case section not passed 1`] = `
<div>
  <div
    data-testid="progress-section__tool-information_error"
    style="display: flex; align-items: center;"
  >
    <div
      class="section-circle"
    >
      <i
        class="eui-icon eui-fa-circle-o invalid-circle"
      />
    </div>
    <div>
      <div
        class="section-label"
      >
        <div
          data-testid="progress-section__tool-information_link"
        >
          Tool Information
        </div>
      </div>
      <div>
        <span
          data-testid="progress-field__name_null"
        >
          <span
            class="field-circle"
            data-testid="progress-field__name_null_required_pass"
          >
            <span
              class="progressField"
              data-testid="progress-field__name_link"
            >
              <i
                class="eui-icon eui-required icon-green pass-required-circle"
                title="Name is required"
              />
            </span>
          </span>
        </span>
        <span
          data-testid="progress-field__long-name_null"
        >
          <span
            class="field-circle"
            data-testid="progress-field__long-name_null_required_error"
          >
            <span
              class="progressField"
              data-testid="progress-field__long-name_link"
            >
              <i
                class="eui-icon eui-fa-minus-circle error-circle"
                title="LongName"
              />
            </span>
          </span>
        </span>
        <span
          data-testid="progress-field__version_null"
        >
          <span
            class="field-circle"
            data-testid="progress-field__version_null_required_pass"
          >
            <span
              class="progressField"
              data-testid="progress-field__version_link"
            >
              <i
                class="eui-icon eui-required icon-green pass-required-circle"
                title="Version is required"
              />
            </span>
          </span>
        </span>
        <span
          data-testid="progress-field__version-description_null"
        >
          <span
            class="field-circle"
            data-testid="progress-field__version-description_null_not-required_not-started"
          >
            <span
              class="progressField"
              data-testid="progress-field__version-description_link"
            >
              <i
                class="eui-icon eui-fa-circle-o not-started-not-required-circle"
                title="VersionDescription"
              />
            </span>
          </span>
        </span>
        <span
          data-testid="progress-field__type_null"
        >
          <span
            class="field-circle"
            data-testid="progress-field__type_null_required_pass"
          >
            <span
              class="progressField"
              data-testid="progress-field__type_link"
            >
              <i
                class="eui-icon eui-required icon-green pass-required-circle"
                title="Type is required"
              />
            </span>
          </span>
        </span>
        <span
          data-testid="progress-field__last-updated-date_null"
        >
          <span
            class="field-circle"
            data-testid="progress-field__last-updated-date_null_not-required_not-started"
          >
            <span
              class="progressField"
              data-testid="progress-field__last-updated-date_link"
            >
              <i
                class="eui-icon eui-fa-circle-o not-started-not-required-circle"
                title="LastUpdatedDate"
              />
            </span>
          </span>
        </span>
        <span
          data-testid="progress-field__description_null"
        >
          <span
            class="field-circle"
            data-testid="progress-field__description_null_required_pass"
          >
            <span
              class="progressField"
              data-testid="progress-field__description_link"
            >
              <i
                class="eui-icon eui-required icon-green pass-required-circle"
                title="Description is required"
              />
            </span>
          </span>
        </span>
        <span
          data-testid="progress-field__doi_null"
        >
          <span
            class="field-circle"
            data-testid="progress-field__doi_null_not-required_pass"
          >
            <span
              class="progressField"
              data-testid="progress-field__doi_link"
            >
              <i
                class="eui-icon eui-fa-circle icon-grey pass-not-required-circle"
                title="DOI"
              />
            </span>
          </span>
        </span>
        <span
          data-testid="progress-field__url_null"
        >
          <span
            class="field-circle"
            data-testid="progress-field__url_null_required_pass"
          >
            <span
              class="progressField"
              data-testid="progress-field__url_link"
            >
              <i
                class="eui-icon eui-required icon-green pass-required-circle"
                title="URL is required"
              />
            </span>
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Progress Section Test renders in case section passed 1`] = `
<div>
  <div
    data-testid="progress-section__tool-information_pass"
    style="display: flex; align-items: center;"
  >
    <div
      class="section-circle"
    >
      <i
        class="eui-icon eui-check pass-circle"
      />
    </div>
    <div>
      <div
        class="section-label"
      >
        <div
          data-testid="progress-section__tool-information_link"
        >
          Tool Information
        </div>
      </div>
      <div>
        <span
          data-testid="progress-field__name_null"
        >
          <span
            class="field-circle"
            data-testid="progress-field__name_null_required_pass"
          >
            <span
              class="progressField"
              data-testid="progress-field__name_link"
            >
              <i
                class="eui-icon eui-required icon-green pass-required-circle"
                title="Name is required"
              />
            </span>
          </span>
        </span>
        <span
          data-testid="progress-field__long-name_null"
        >
          <span
            class="field-circle"
            data-testid="progress-field__long-name_null_required_pass"
          >
            <span
              class="progressField"
              data-testid="progress-field__long-name_link"
            >
              <i
                class="eui-icon eui-required icon-green pass-required-circle"
                title="LongName is required"
              />
            </span>
          </span>
        </span>
        <span
          data-testid="progress-field__version_null"
        >
          <span
            class="field-circle"
            data-testid="progress-field__version_null_required_pass"
          >
            <span
              class="progressField"
              data-testid="progress-field__version_link"
            >
              <i
                class="eui-icon eui-required icon-green pass-required-circle"
                title="Version is required"
              />
            </span>
          </span>
        </span>
        <span
          data-testid="progress-field__version-description_null"
        >
          <span
            class="field-circle"
            data-testid="progress-field__version-description_null_not-required_not-started"
          >
            <span
              class="progressField"
              data-testid="progress-field__version-description_link"
            >
              <i
                class="eui-icon eui-fa-circle-o not-started-not-required-circle"
                title="VersionDescription"
              />
            </span>
          </span>
        </span>
        <span
          data-testid="progress-field__type_null"
        >
          <span
            class="field-circle"
            data-testid="progress-field__type_null_required_pass"
          >
            <span
              class="progressField"
              data-testid="progress-field__type_link"
            >
              <i
                class="eui-icon eui-required icon-green pass-required-circle"
                title="Type is required"
              />
            </span>
          </span>
        </span>
        <span
          data-testid="progress-field__last-updated-date_null"
        >
          <span
            class="field-circle"
            data-testid="progress-field__last-updated-date_null_not-required_not-started"
          >
            <span
              class="progressField"
              data-testid="progress-field__last-updated-date_link"
            >
              <i
                class="eui-icon eui-fa-circle-o not-started-not-required-circle"
                title="LastUpdatedDate"
              />
            </span>
          </span>
        </span>
        <span
          data-testid="progress-field__description_null"
        >
          <span
            class="field-circle"
            data-testid="progress-field__description_null_required_pass"
          >
            <span
              class="progressField"
              data-testid="progress-field__description_link"
            >
              <i
                class="eui-icon eui-required icon-green pass-required-circle"
                title="Description is required"
              />
            </span>
          </span>
        </span>
        <span
          data-testid="progress-field__doi_null"
        >
          <span
            class="field-circle"
            data-testid="progress-field__doi_null_not-required_pass"
          >
            <span
              class="progressField"
              data-testid="progress-field__doi_link"
            >
              <i
                class="eui-icon eui-fa-circle icon-grey pass-not-required-circle"
                title="DOI"
              />
            </span>
          </span>
        </span>
        <span
          data-testid="progress-field__url_null"
        >
          <span
            class="field-circle"
            data-testid="progress-field__url_null_required_pass"
          >
            <span
              class="progressField"
              data-testid="progress-field__url_link"
            >
              <i
                class="eui-icon eui-required icon-green pass-required-circle"
                title="URL is required"
              />
            </span>
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Progress Section Test renders in case section status unknown 1`] = `
<div>
  <div
    data-testid="progress-section__tool-contacts_made-up"
    style="display: flex; align-items: center;"
  >
    <div
      class="section-circle"
    >
      <span>
        Status Unknown
      </span>
    </div>
    <div>
      <div
        class="section-label"
      >
        <div
          data-testid="progress-section__tool-contacts_link"
        >
          Tool Contacts
        </div>
      </div>
      <div>
        <span
          data-testid="progress-field__contact-groups_null"
        >
          <span
            class="field-circle"
            data-testid="progress-field__contact-groups_null_required_not-started"
          >
            <span
              class="progressField"
              data-testid="progress-field__contact-groups_link"
            >
              <i
                class="eui-icon eui-required-o not-started-required-circle"
                title="ContactGroups is required"
              />
            </span>
          </span>
        </span>
        <span
          data-testid="progress-field__contact-persons_null"
        >
          <span
            class="field-circle"
            data-testid="progress-field__contact-persons_null_not-required_not-started"
          >
            <span
              class="progressField"
              data-testid="progress-field__contact-persons_link"
            >
              <i
                class="eui-icon eui-fa-circle-o not-started-not-required-circle"
                title="ContactPersons"
              />
            </span>
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
